workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/SendCreatePlayerRequestSystem.cs

```diff
@@ -44,25 +53,35 @@ protected override void OnCreateManager(int capacity)
 
         protected override void OnUpdate()
         {
+            for (int i = 0; i < newEntityData.Length; ++i)
+            {
+                PostUpdateCommands.AddComponent(newEntityData.Entities[i], new ShouldRequestPlayerTag());
+            }
+
             for (var i = 0; i < sendData.Length; ++i)
             {
                 var request = new CreatePlayerRequestType(new Generated.Improbable.Vector3f { X = 0, Y = 0, Z = 0 });
                 var createPlayerRequest = PlayerCreator.CreatePlayer.CreateRequest(playerCreatorEntityId, request);
 
                 sendData.RequestSenders[i].RequestsToSend
                     .Add(createPlayerRequest);
+                PostUpdateCommands.RemoveComponent<ShouldRequestPlayerTag>(sendData.Entities[i]);
             }
 
+            // Currently this has a race condition where you can receive two entites
+            // The fix for this is more sophisticted server side handling of requests
             for (var i = 0; i < responseData.Length; ++i)
             {
                 foreach (var receivedResponse in responseData.Responses[i].Responses)
                 {
-                    var requestId = receivedResponse.RequestId;
-
-                    if (receivedResponse.StatusCode != StatusCode.Success)
+                    if (receivedResponse.StatusCode == StatusCode.AuthorityLost)
+                    {
+                        PostUpdateCommands.AddComponent(responseData.Entities[i], new ShouldRequestPlayerTag());
```

Comment:
We want those to still fail for the moment. Otherwise we introduce races where you can end up with two players. We can solve that properly but this is the minimal fix for the end of day.